src/overlays/actors/ovl_En_Demo_heishi/z_en_demo_heishi.c

```diff
@@ -31,37 +36,149 @@ const ActorInit En_Demo_heishi_InitVars = {
+    this->frameCount = Animation_GetLastFrame(sAnimations[animIndex]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.0f, this->frameCount,
+                     sAnimModes[this->animIndex], -10.0f);
+}
+
+void EnDemoheishi_SetHeadRotation(EnDemoheishi* this) {
+    s16 yawTemp = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    s32 yaw = ABS_ALT(yawTemp);
+
+    this->headRotXTarget = 0;
+    if ((this->actor.xzDistToPlayer < 200.0f) && (yaw < 20000)) {
+        this->headRotXTarget = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+        if (this->headRotXTarget > 10000) {
+            this->headRotXTarget = 10000;
+        } else if (this->headRotXTarget < -10000) {
+            this->headRotXTarget = -10000;
+        }
+    }
+}
+
+void EnDemoheishi_SetupIdle(EnDemoheishi* this) {
+    EnDemoheishi_ChangeAnimation(this, 0);
+    this->unk272 = 0;
+    this->actor.textId = sTextIds[this->unk272];
+    this->unk270 = 0;
+    this->actionFunc = EnDemoheishi_Idle;
+}
+
+void EnDemoheishi_Idle(EnDemoheishi* this, GlobalContext* globalCtx) {
+    s32 absYawDiff;
+    s16 yawDiff;
+
+    this->actor.flags &= ~0x8000000;
+    yawDiff = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    absYawDiff = ABS_ALT(yawDiff);
+
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        EnDemoheishi_SetupTalk(this);
+    } else if (absYawDiff <= 0x4BB8) {
+        func_800B8614(&this->actor, globalCtx, 70.0f);
+    }
+}
+
+void EnDemoheishi_SetupTalk(EnDemoheishi* this) {
+    this->unk270 = 1;
+    this->actionFunc = EnDemoheishi_Talk;
+}
+
+void EnDemoheishi_Talk(EnDemoheishi* this, GlobalContext* globalCtx) {
+    if ((Message_GetState(&globalCtx->msgCtx) == 5) && (func_80147624(globalCtx) != 0)) {
+        func_801477B4(globalCtx);
+        EnDemoheishi_SetupIdle(this);
+    }
+}
+
+void EnDemoheishi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnDemoheishi* this = THIS;
+
+    SkelAnime_Update(&this->skelAnime);
+    if (this->unk268 != 0) {
+        this->unk268--;
+    }
+
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
+    Actor_SetScale(&this->actor, 0.01f);
+    EnDemoheishi_SetHeadRotation(this);
+
+    Actor_SetFocus(&this->actor, 60.0f);
+    Math_SmoothStepToS(&this->headRotX, this->headRotXTarget, 1, 0xBB8, 0);
+    Math_SmoothStepToS(&this->headRotY, this->headRotYTarget, 1, 0x3E8, 0);
+    Collider_UpdateCylinder(&this->actor, &this->colliderCylinder);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder.base);
+}
+
+s32 EnDemoheishi_OverrideLimbDraw(GlobalContext* globalctx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                  Actor* thisx) {
+    EnDemoheishi* this = THIS;
+
+    if (limbIndex == 16) {
```

Comment:
When documenting the object, we prefer if peaple make an enum to match the limbs. Beware that they'll be off by one from the numbers in the object, but in the same order. So this will be `DEMOHEISHI_LIMB_HEAD` or similar. If you search for `_LIMB_` in the codebase you'll find some examples, although OoT has many more than we currently do since we only started objects recently.

Reply:
Fixed